pkg/validation/spec/header.go

```diff
@@ -62,6 +62,43 @@ func (h Header) MarshalJSON() ([]byte, error) {
 	return swag.ConcatJSON(b1, b2, b3, b4), nil
 }
 
+func (h Header) MarshalNextJSON(opts jsonv2.MarshalOptions, enc *jsonv2.Encoder) error {
+	type SimpleSchemaOmitZero struct {
```

Comment:
There's a lot of repetition in these types (SimpleSchemaOmitZero is defined at least a couple of times), and even the `x` structs are re-use between MarshalNext and UnmarshalNext. Should we name them (uncapitalized) next to the types they copy so that we can re-use?

Reply:
`omitzero` is ignored by the v1 "json" marshaler, we could just put those on original types. That will probably remove most of the duplication.

Reply:
Actually, I copied these types because I couldn't get a combination that behaves the same way with both v1 and v2.
v1 requires omitempty, v2 requires omitzero without omitempty.

Reply:
Couldn't you put both "omitzero,omitempty" for some set of these? The "omitzero" will still be ignored by v1, but that's okay. For example, that would work for all the cases with boolean and number types.

Reply:
Yeah, for boolean and number types, I only needed to add omitzero to existing types. I had to copy the whole thing to have omitzero without omitempty for pointers to structs (since omitempty is now removing structs that are serialized to `{}`)

Reply:
Ah, got it. I'll make a note of this for v1-to-v2 compatibility issues.

Reply:
To be honest, it'd have been very nice if we could get a way to have the flags to give me something compatible. I basically wrote something like this: https://go.dev/play/p/4yckWam8Pw1 to try to find a combination that works with both, but there was no way to get the same output in both v1 and v2 without duplicating the structure.

Reply:
It's possible we'll add flags to v2 to have some more v1-like behavior. Getting the v1-to-v2 compatibility story is possibly the hardest part of the puzzle.

Reply:
Using reflection and all? Between the compilation failing and the round-trip tests, I don't see this as necessary nor useful.

Reply:
I thought we might be vulnerable to case-insensitivity in the json tag names, but it looks like the tests are comparing the actual json marshaled bytes, so I agree.

package-private `___OmitZero` types next to the existing types _would_ let you avoid declaring `SimpleSchemaOmitZero` and `CommonValidationsOmitZero` three times